### ISSUE TITLE
Some apps cannot stage with 'Error: App has not finished staging'

### ISSUE THREAD

**Issue by Darren-Vine** [cloudfoundry-community/terraform-provider-cloudfoundry#379]
An app always deploys successfully first time.  However if the docker image changes some apps report 

```
Error: App has not finished staging
  with [module.cf](http://module.cf/)_aicloud_apps.cloudfoundry_app.widget,
  on ../../modules/cf_aicloud_apps/a_widget.tf line 11, in resource "cloudfoundry_app" "widget":
  11: resource "cloudfoundry_app" "widget" {
```
  
when terraform says

```
~ resource "cloudfoundry_app" "widget" {
      ~ docker_image               = "[xxx/widget:0.1.36084](xxx/widget:0.1.36084)" -> "[xxx/widget:0.1.36000-widgetconsumed](xxx/widget:0.1.36000-widgetconsumed)"
      ~ environment                = (sensitive value)
        id                         = "54934d9d-eb90-4923-82ea-0e8b8ec00d68"
      ~ id_bg                      = "54934d9d-eb90-4923-82ea-0e8b8ec00d68" -> (known after apply)
        name                       = "widget-devops01"
        # (18 unchanged attributes hidden)
```

If we just rerun then it will pass successfully but cf app still says it is running on the old image. Terraform state however seems to think it is on the updated container.

```
name:              widget-devops01
requested state:   started
routes:            widget-devops01.apps.internal
last uploaded:     Tue 22 Feb 15:39:32 CET 2022
stack:
docker image:      [xxx/widget:0.1.36084](xxx/widget:0.1.36084)
```

Only deleting the app will update it and redeploying will update this image.  We need this resolved as we have no clue what is actually running and developers are complaining that their changes are not being deployed.  Interestingly it is mostly only happening with a couple of apps.  Others update without issues.

**Comment by loafoe** [comment]
I did some investigation on this and when this happens the `v2` API indeed returns the new image. Currently the provides uses this as the source of truth. When using the `v3` API, specifically the droplet endpoint you will still see the old image, which is also what the newer CLI versions use. I'm making a minor change to the CF app state reading to now also sync against the Droplet endpoint. 
This should at least let the provider detect and re-deploy the app on a subsequent run. The proper fix is to migrate completely to the `v3` API but this is a herculean task waiting to be picked up.

**Comment by khalilj** [comment]
@Darren-Vine Have you tested this fix and can confirm it resolves the issue?

**Comment by Darren-Vine** [comment]
@khalilj No, we implemented a work around and we havent come back to this due to time contraints.

**Comment by loafoe** [comment]
@Darren-Vine please test with the `rolling` strategy

**Comment by durga61** [comment]
@khalilj , could you please share workaround for this, been facing this error for sometime
